### PR TITLE
Add @bonadocs/docgen

### PR DESCRIPTION
**Pull request by staa99**
- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [ ] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.
- [x] I didn't do anything of this.

---

Add `@bonadocs/docgen` to the plugins directory. It enables users to generate an interactive documentation website featuring the bonadocs widget that allows developers to interact with the smart contracts directly from within the documentation.

### CODE REVIEW

**Comment by fvictorio** [comment]
HI @staa99, can you run prettier on the file you modified? Thanks.

File: docs/src/content/hardhat-runner/plugins/plugins.ts [hunk 1]
```diff
@@ -863,6 +863,13 @@ const communityPlugins: IPlugin[] = [
       "Display different signatures that have the methods, events and errors of your contracts by console",
     tags: ["functions", "errors", "events", "tooling", "selectorss"],
   },
+  {
```
Comment (schaable) [inline docs/src/content/hardhat-runner/plugins/plugins.ts:866]: you're missing the `npmPackage` key.

Comment (staa99) [reply docs/src/content/hardhat-runner/plugins/plugins.ts:866]: Thanks @schaable

**Comment by staa99** [comment]
I closed the PR to rebase the branch on the latest to avoid conflicts. New PR is #4810

**Pull request closed without merging**